src/plugins/intel_gpu/src/graph/compilation_context.cpp

```diff
@@ -40,11 +48,32 @@ class CompilationContext : public ICompilationContext {
 
     ~CompilationContext() noexcept { cancel(); }
 
+private:
+    bool get_front_task(ICompilationContext::Task& task) {
+        std::lock_guard<std::mutex> lock(_mutex);
+        if (!_compile_queue.empty()) {
+            auto front = _compile_queue.front();
+            task = front.second;
+            return true;
+        }
+        return false;
+    }
+
+    void pop_front_task() {
+        std::lock_guard<std::mutex> lock(_mutex);
+        auto front = _compile_queue.front();
+        _compile_queue_keymap.erase(front.first);
```

Comment:
As I asked offline, erasing this from keymap as soon as pop, the duplicate enqueue might occur while the same task is still being compiled. 
So I think erasing should happen afther the compilation is done.

Reply:
pop_front_task() is called after compilation is done. so it is erased and I think it is working as we expected.
Could you please check it again?

Reply:
Change function name from pop_front_task to erase_front_task.

Reply:
But, shouldnt the popping from the queue itself be called right after the task began to executed? 
Otherwise, other thread can get the task from the compilation queue.
keymap should be erased after the compilation finished to prevent pushing same task. 
However removing from the queue should be done as soon as the compilation is begin.. ?

Reply:
Update compilation queue. please review it again. thanks.